src/services/item/ws/services.ts

```diff
@@ -0,0 +1,115 @@
+import { FastifyBaseLogger } from 'fastify';
+
+import { getParentFromPath } from '@graasp/sdk';
+
+import { WebsocketService } from '../../websockets/ws-service';
+import { Item } from '../entities/Item';
+import { getPromiseResults } from '../utils';
+import {
+  AccessibleItemsEvent,
+  ChildItemEvent,
+  ItemOpFeedbackEvent,
+  OwnItemsEvent,
+  itemTopic,
+  memberItemsTopic,
+} from './events';
+
+// on move item:
+// - notify own items of creator of delete IF old location was root
+// - notify own items of creator of create IF new location is root
+const publishMemberItemsTopic = (
+  websockets: WebsocketService,
+  { source, destination, sourceParentId },
+) => {
+  if (sourceParentId === undefined && source.creator) {
+    // root item, notify creator
+    // todo: remove own when we don't use own anymore
+    websockets.publish(memberItemsTopic, source.creator.id, OwnItemsEvent('delete', source));
+    websockets.publish(memberItemsTopic, source.creator.id, AccessibleItemsEvent('delete', source));
+  }
+
+  const destParentId = getParentFromPath(destination.path);
+  if (destParentId === undefined && destination.creator) {
+    // root item, notify creator
+    // todo: remove own when we don't use own anymore
+    websockets.publish(
+      memberItemsTopic,
+      destination.creator.id,
+      OwnItemsEvent('create', destination),
+    );
+    websockets.publish(
+      memberItemsTopic,
+      destination.creator.id,
+      AccessibleItemsEvent('create', destination),
+    );
+  }
+};
+
+// on move item, notify:
+// - parent of old location of deleted child
+// - parent of new location of new child
+const publishItemTopic = (
+  websockets: WebsocketService,
+  { source, destination, sourceParentId },
+) => {
+  if (sourceParentId !== undefined) {
+    websockets.publish(itemTopic, sourceParentId, ChildItemEvent('delete', source));
+  }
+
+  const destParentId = getParentFromPath(destination.path);
+  if (destParentId) {
+    websockets.publish(itemTopic, destParentId, ChildItemEvent('create', destination));
+  }
+};
+
+export const publishAfterMoved = (
```

Comment:
I don't really like the name. Actually if you create a class to group functions (later copy, update, etc) you won't need to pass the websocket service everytime.

```ts
class ItemWebsocketsService {
   constructor(websockets: WebsocketService) { ... }

   publishTopicsForMove({ source, destination, sourceParentId }) { ... }

   publishFeedbacksForMove(...) { ... }
}
```
and use 
```ts
const itemWebsocketsService = new ItemWebsocketsService(...)

itemWebsocketsService.publishTopicsForMove()
```

Reply:
Yes, I started to do something similar in copy. It's a good idea, I will add it in move PR too.